lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -240,19 +240,21 @@ bool CJsonStateRestoreTraverser::parseNext(bool remember) {
 
         m_Handler.s_RememberValue = remember;
 
-        char c = '\0';
-        m_ReadStream.get(c);
-        if (c == '\0') {
-            break;
+        if (m_BytesRemaining == 0) {
+            m_BytesRemaining = BUFFER_SIZE - 1;
+            ::memset(m_Buffer, '\0', BUFFER_SIZE);
+            m_ReadStream.get(m_Buffer, BUFFER_SIZE, '\0');
+            m_BufferPtr = m_Buffer;
         }
 
         json::error_code ec;
-        m_Reader.write_some(true, &c, 1, ec);
+        std::size_t written = m_Reader.write_some(true, m_BufferPtr++, 1, ec);
         if (ec) {
             this->logError();
             ret = false;
             break;
         }
+        m_BytesRemaining -= written;
```

Comment:
By modifying `m_BytesRemaining` after the error handling but `m_BufferPtr` before, they could be out-of-sync next time the method is called. Is this a problem? Even if it's not now it might be better to modify them together to avoid pitfalls for future maintainers.